src/pageLayout/containers/MainNavigation.tsx

```diff
@@ -9,37 +9,226 @@ import UserWidget from 'src/pageLayout/components/UserWidget'
 import NavHeader from 'src/pageLayout/components/NavHeader'
 
 // Constants
-import {generateNavItems} from 'src/pageLayout/constants/navigationHierarchy'
 import {CLOUD} from 'src/shared/constants'
 
 // Utils
 import {getNavItemActivation} from 'src/pageLayout/utils'
 import {AppSettingContext} from 'src/shared/contexts/app'
 import {event} from 'src/cloud/utils/reporting'
 import {SafeBlankLink} from 'src/utils/SafeBlankLink'
+import {isFlagEnabled} from 'src/shared/utils/featureFlag'
+import {isUserOperator} from 'src/operator/utils'
 
 // Selectors
 import {getOrg} from 'src/organizations/selectors'
 import {
   selectCurrentAccountType,
   selectOperatorRole,
 } from 'src/identity/selectors'
+import {selectShouldShowNotebooks} from 'src/flows/selectors/flowsSelectors'
 
 // Types
-import {NavItem, NavSubItem} from 'src/pageLayout/constants/navigationHierarchy'
+import {IdentityUser} from 'src/client/unityRoutes'
 
 // Overlays
 import {showOverlay, dismissOverlay} from 'src/overlays/actions/overlays'
 
 // Styles
 import './MainNavigation.scss'
 
+interface NavItem {
+  id: string
+  testID: string
+  label: string
+  shortLabel?: string
+  link: string
+  icon: IconFont
+  enabled?: () => boolean
+  menu?: NavSubItem[]
+  activeKeywords: string[]
+}
+
+interface NavSubItem {
+  id: string
+  testID: string
+  label: string
+  link: string
+  enabled?: () => boolean
+}
+
+const generateNavItems = (
```

Comment:
I moved this into here from `navigationHierarchy`. That was an example of a function that was in a different file that was used in a single place once.

Reply:
In fact, I spun this into a separate PR. Review that one first: https://github.com/influxdata/ui/pull/6423